### PR TITLE
Add swap function.

### PR DESCRIPTION
**Pull request by adrianb93**
This function swaps the values of two variables.

```php
$startDate = '2040-01-01';
$endDate = '2020-01-01';

if ($endDate < $startDate) {
    swap($startDate, $endDate);
}

echo $startDate; // prints "2020-01-01"
echo $endDate; // prints "2040-01-01"
```

### CODE REVIEW

File: src/helpers/swap.php [hunk 1]
```diff
@@ -0,0 +1,8 @@
+<?php
+
+function swap(&$a, &$b)
+{
+    $temp = $a;
+    $a = $b;
+    $b = $temp;
```
Comment (CupOfTea696) [inline src/helpers/swap.php]: Given that Laravel 5.7+ only supports PHP 7.1+, the entire body of this function can be simplified to `[$a, $b] = [$b, $a];`

Comment (adrianb93) [reply src/helpers/swap.php]: I actually googled it before I did this PR. Someone did benchmarks and it's around twice as slow, so I figured I'd just be old fashioned.

https://stackoverflow.com/questions/3541730/is-there-a-php-function-for-swapping-the-values-of-two-variables

```
<?php

function stopwatch($callback, $times = 1)
{
    $totalTime = 0;

    foreach (range(1, $times) as $time) {
        $start = microtime(true);
        $callback();
        $totalTime += microtime(true) - $start;
    }

    return $totalTime / $times;
}

function swap1(&$a, &$b)
{
	[$a, $b] = [$b, $a];
}

function swap2(&$a, &$b)
{
	$temp = $a;
	$b = $a;
	$a = $temp;
}

echo stopwatch(function () { $a = 1; $b = 2; swap1($a, $b); }, 100) . PHP_EOL;
echo stopwatch(function () { $a = 1; $b = 2; swap2($a, $b); }, 100) . PHP_EOL;
// Result:
// 5.6266784667969E-7
// 2.0980834960938E-7
```

Comment (CupOfTea696) [reply src/helpers/swap.php]: Fair enough, then.

**Comment by calebporzio** [comment]
This is a pretty cool lil' function. Two thoughts:

A) is it obvious enough what it does? I think the answer is yes. After all "tap" isn't obvious at all, it just becomes an idiom.
B) what are some other use cases. I can remember needing something like this, just want to make sure it's truly useful, and not just fun to look at.
C) will this be confused with Laravel's Facade::swap? When I first saw the name of the function, I thought it was a helper to swap instances in the container.

I'm truly not opposed to this, just want to talk some things out.

**Comment by akiyamaSM** [comment]
For the (c) it would be helpful if the name is changed to var_swap()

**Comment by adrianb93** [comment]
A) I think it's obvious too, it reads really well.
B) I've only ever used it for sorting function parameters for range related things. For example, a period of time class.
C) I just learned what facade swap is. If I was to swap the implementation behind a facade, I think it would be clearer intent to do so through the facade:

```php
Auth::swap(
    resolve(OperatingSystemAuth::class)
);
```

I don't think that's the type of thing I'd reach for a helper for. I'm open to naming suggestions. I like how `swap` reads, and how it's short and sweet though.

**Comment by adrianb93** [comment]
Just wanted to give this another bump. I found another use case for `swap()`. I was writing this helper function:

```php
function abort_if_caught($code, $exception, $callback = null)
{
    if ($callback === null) {
        swap($exception, $callback);
    }

    try {
        return $callback();
    } catch (Exception $error) {
        if ($exception && $error instanceof $exception === false) {
            throw $error;
        }

        abort($code);
    }
}
```

I wanted the `$excpetion` parameter to be optional, but the `$callback` to be required. If no third parameter was provided, I used the swap function to swap the values of `$exception` and `$callback`.

**Comment by calebporzio** [comment]
Good use case - let's do it!

**Pull request merged**